src/app/shared/cms/haldoc.ts

```diff
@@ -0,0 +1,97 @@
+import {Http, Headers, RequestOptions, Response} from 'angular2/http';
+import {Observable} from 'rxjs/Observable';
+
+/*
+ * Generic class for interacting with HAL api
+ */
+export class HalDoc {
+  private _http: Http;
```

Comment:
I dig the `_privatevar` naming, but it's inconsistent. I think since ts will check this for us, it might be ok to leave it off, which could also make the constructor simpler if the args declared with private properties.